### ISSUE TITLE
TS generates "use strict" in front of everything, this makes impossible to use its output for Task Runner Explorer without touching the output manually (or by script)

### ISSUE THREAD

**Issue by fis-cz** [microsoft/TypeScript#15819]
**TypeScript Version:**  2.2.1 - 2.9.1

TS generates the "use strict" before starting notes in the js file. These notes can include directions for post-processing tools, or another tools like Task Runner Explorer in Visual Studio which looks for the binding information on the first line of the gulpfile.js or gruntfile.js. It is necessary to write a post-build task to fix this and makes it impossible to include the binding information directly to ts file if gulpfile or gruntfile is written in typescript.

**Code**

```
﻿/// <binding Clean='clean' ProjectOpened='projectOpen' />
/* some notes */
function test(): void {
}
```

**Expected behavior:**

```
﻿/// <binding Clean='clean' ProjectOpened='projectOpen' />
/* some notes */
"use strict";
function test() {
}
```

**Actual behavior:**

```
"use strict";
﻿/// <binding Clean='clean' ProjectOpened='projectOpen' />
/* some notes */
function test() {
}
```

**Comment by DanielRosenwasser** [comment]
As a work-around, you can use `noImplicitUseStrict` to get rid of the generated prologue.

**Comment by fis-cz** [comment]
@DanielRosenwasser Nice tip, thanks, at this time I am good to go with post-build task.

**Comment by Baxterboom** [comment]
Howdy! 
In regards to **noImplicitUseStrict**, i cant get it to work with this configuration...

`{
	"compileOnSave": true,
	"compilerOptions": {
		"noImplicitUseStrict": true,
		"outFile": "scripts/typescript.generated.bundle.js",
		"removeComments": true,
		"sourceMap": false,
		"target": "es5"
	}
}
`
... its always added to top of file...

`'use strict';
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();`

... is there something i am missing?

**Comment by DanielRosenwasser** [comment]
@Baxterboom are you still running into this with TypeScript 2.4.2? I'm not able to repro it.

**Comment by Baxterboom** [comment]
Ill look into it once i get back from vacation.

**Comment by fis-cz** [comment]
Just want to note this is still valid for 2.9.1.

**Comment by pchodak** [comment]
@fis-cz did you manage to fix this?

**Comment by fis-cz** [comment]
@pchodak  Yes. I did. As well as many other problems with typescript which are not fixed or not wanted to be implemented. I moved to "es next" and I am using babel to transpile to ES5 if I need.

**Comment by wilsonianb** [comment]
I'm currently encountering this while trying to use [@jest-environment docblock](https://jestjs.io/docs/en/configuration#testenvironment-string) at the top of files with TypeScript version 3.6.3

**Comment by bhitov** [comment]
@wilsonianb 

Your comment is from 2019 so I doubt this is still an issue for you, but I ran into the same thing and this is the workaround I found:

`index.test.ts`
```typescript
/** @jest-environment jsdom */
// eslint-disable-next-line strict, lines-around-directive
'use strict';
// use strict is necessary here so that typescript doesn't place "use strict" above the jest docblock
```

which transpiles to:

`index.test.js`
```javascript
/** @jest-environment jsdom */
// eslint-disable-next-line strict, lines-around-directive
'use strict';
```

It seems that by adding your own explicit `'use strict'` you can prevent tsc from placing it at the top of the emitted file.

**Comment by RyanCavanaugh** [comment]
Closing as Won't Fix - the output isn't wrong, it's just other tools failing to handle a legal setup. Plus this only matters for global code (module code is implicitly `use strict`) which very few people are writing anymore
